master/internal/api_trials.go

```diff
@@ -682,60 +698,32 @@ func (a *apiServer) MultiTrialSample(trialID int32, metricNames []string,
 	if endBatches == 0 {
 		endBatches = math.MaxInt32
 	}
+	if maxDatapoints == 0 {
+		maxDatapoints = 200
+	}
 
-	if len(metricNames) > 0 {
-		if (metricType == apiv1.MetricType_METRIC_TYPE_TRAINING) ||
-			(metricType == apiv1.MetricType_METRIC_TYPE_UNSPECIFIED) {
-			var metric apiv1.DownsampledMetrics
-			metricMeasurements, err = trials.MetricsTimeSeries(
-				trialID, startTime, metricNames, startBatches, endBatches,
-				xAxisLabelMetrics,
-				maxDatapoints, batches, timeSeriesFilter, "training")
-			if err != nil {
-				return nil, errors.Wrapf(err, "error fetching time series of training metrics")
-			}
-			metric.Type = apiv1.MetricType_METRIC_TYPE_TRAINING
-			if len(metricMeasurements) > 0 {
-				if err = a.formatMetrics(&metric, metricMeasurements); err != nil {
-					return nil, err
-				}
-				metrics = append(metrics, &metric)
-			}
-		}
-
-		if (metricType == apiv1.MetricType_METRIC_TYPE_VALIDATION) ||
-			(metricType == apiv1.MetricType_METRIC_TYPE_UNSPECIFIED) {
-			var metric apiv1.DownsampledMetrics
-			metricMeasurements, err = trials.MetricsTimeSeries(
-				trialID, startTime, metricNames, startBatches, endBatches,
-				xAxisLabelMetrics, maxDatapoints, batches, timeSeriesFilter, "validation")
-			if err != nil {
-				return nil, errors.Wrapf(err, "error fetching time series of validation metrics")
-			}
-			metric.Type = apiv1.MetricType_METRIC_TYPE_VALIDATION
+	var timeSeriesColumn *string
 
-			if len(metricMeasurements) > 0 {
-				if err = a.formatMetrics(&metric, metricMeasurements); err != nil {
-					return nil, err
-				}
-				metrics = append(metrics, &metric)
-			}
-		}
+	// If no time series filter column name is supplied then default to batches.
+	defaultTimeSeriesColumn := batches
+	if timeSeriesFilter == nil || timeSeriesFilter.Name == nil {
+		timeSeriesColumn = &defaultTimeSeriesColumn
+	} else {
+		timeSeriesColumn = timeSeriesFilter.Name
 	}
 
-	if len(metricIds) > 0 {
-		var timeSeriesColumn *string
-
-		// If no time series filter column name is supplied then default to batches.
-		defaultTimeSeriesColumn := batches
-		if timeSeriesFilter == nil || timeSeriesFilter.Name == nil {
-			timeSeriesColumn = &defaultTimeSeriesColumn
+	// TODO: we could throw out duplicates. (breaking change?)
```

Comment:
I doubt having duplicates has ever been intentional. don't think this qualifies as "breaking"